node/src/Pos/Wallet/Web/Pending/Updates.hs

```diff
@@ -39,3 +41,27 @@ ptxMarkAcknowledgedPure :: PendingTx -> PendingTx
 ptxMarkAcknowledgedPure = execState $ do
     wasAcked <- ptxPeerAck <<.= True
     unless wasAcked $ ptxSubmitTiming . pstNextDelay %= (* 8)
+
+-- | If given transaction has been canceled, sets its condition
+-- to 'PtxApplying'. This allows "stuck" transactions to be resubmitted
+-- again.
+--
+-- Has no effect for transactions in other conditions.
+resetFailedPtx :: HasConfiguration => SlotId -> PendingTx -> PendingTx
```

Comment:
We explicitly ignored this piece of work, originally available [here](https://github.com/input-output-hk/cardano-sl/pull/2213/commits/559676c16ecebe2606f347b6cbd9a6e3fe45a3ed). We decided not to port it.